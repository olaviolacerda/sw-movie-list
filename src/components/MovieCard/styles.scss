.card {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  padding: 1em;
  background: #0c0c0c;
  border-radius: 8px;
  position: relative;
  border: 1px solid;
  min-height: 300px;
  width: 100%;

  &:hover {
    box-shadow: 0 0 5px 1px #edec51;
    border: 1px solid #fff;
  }

  .title {
    color: #fff;
    font-weight: bold;
    font-size: 36px;
  }

  .date {
    font-weight: bold;
    color: #514f4f;
  }
}
