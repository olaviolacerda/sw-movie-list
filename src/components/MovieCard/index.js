import React from 'react';
import {Link} from 'react-router-dom';
import Button from '../Button';

import './styles.scss';

export default function MovieCard({movie}) {
  const {title, release_date: date} = movie;
  return (
    <div className="card">
      <h5 className="title">{title}</h5>
      <p className="date">{date}</p>
      <Button>
        <Link to={`/movies/${movie.episode_id}/details`}>See More</Link>
      </Button>
    </div>
  );
}
