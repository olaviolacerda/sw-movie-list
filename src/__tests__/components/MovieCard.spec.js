import React from 'react';
import {shallow} from 'enzyme';
import MovieCard from '../../components/MovieCard';

describe('Movie Card component', () => {
  it('should render without crashing', () => {
    shallow(<MovieCard movie={{}} />);

    expect(MovieCard).toBeDefined();
  });

  it('should render correctly', () => {
    const movieCardElement = shallow(<MovieCard movie={{}} />);

    expect(movieCardElement).toMatchSnapshot();
  });
});
