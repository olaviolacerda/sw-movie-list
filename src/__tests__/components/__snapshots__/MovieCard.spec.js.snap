// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Card component should render correctly 1`] = `
<div
  className="card"
>
  <h5
    className="title"
  />
  <p
    className="date"
  />
  <Link
    className="button"
    to="/movies/undefined/details"
  >
    Details
  </Link>
</div>
`;
